test/core.jl

```diff
@@ -3140,6 +3140,38 @@ end
 @test_throws TypeError MyType8010([3.0;4.0])
 @test_throws TypeError MyType8010_ghost([3.0;4.0])
 
+module TestNewTypeError
+using Base.Test
+
+struct A
+end
+struct B
+    a::A
+end
+@eval function f1()
+    # Emitting this direction is not recommended but it can come from `convert` that does not
+    # return the correct type.
+    $(Expr(:new, B, 1))
+end
+@eval function f2()
+    a = $(Expr(:new, B, 1))
+    a = a
+    return nothing
+end
+@generated function f3()
+    quote
+        # This is currently broken since the expression is removed by the frontend.
```

Comment:
I think it might be unclear that these tests are even valid syntax. It's not a valid AST construct (cannot be produced by the parser), so I think it could be considered an accident that lowering accepts this.